karibu-dsl/src/main/kotlin/com/github/mvysny/karibudsl/v10/KLitRendererDsl.kt

```diff
@@ -0,0 +1,166 @@
+ * FAIL: $\{item.itemName}
+ * --- Templates escaping in Kotlin multiline strings https://stackoverflow.com/questions/32993586/templates-escaping-in-kotlin-multiline-strings
+ * Solution: ${ "\${item.itemName}" }
+ */
+public val String.litItem get() = "\${item.$this}"
+
+@Suppress("unused")
+@VaadinDsl
+public interface KLitRendererBuilderA<TSource> {
+
+    public data class Property<TSource>(
+        val name: String,
+        val provider: (TSource) -> String,
+    ) {
+        public val litItem: String get() = name.litItem
+    }
+
+    /**
+     * --- withFunction https://vaadin.com/api/platform/com/vaadin/flow/data/renderer/LitRenderer.html#withFunction(java.lang.String,com.vaadin.flow.function.SerializableConsumer)
+     * @param name This name must be a valid JavaScript function name. It must be alphanumeric and not null, must not be one of the JavaScript reserved words (https://www.w3schools.com/js/js_reserved.asp)
+     */
+    public data class Function<TSource>(
+        val name: String,
+        val handler: (TSource) -> Unit,
+    ) {
+
+        /**
+         * NOTE: Rules to validate name of function are created by ChatGPT :)
+         */
+        private companion object {
+            // JavaScript identifier rules
+            val identifierRegex = Regex("^[a-zA-Z_\$][a-zA-Z0-9_\$]*$")
+
+            // Reserved keywords in JavaScript
+            val reservedKeywords = setOf(
+                "break", "case", "catch", "class", "const", "continue", "debugger", "default",
+                "delete", "do", "else", "enum", "export", "extends", "false", "finally",
+                "for", "function", "if", "import", "in", "instanceof", "new", "null", "return",
+                "super", "switch", "this", "throw", "true", "try", "typeof", "var", "void",
+                "while", "with", "yield", "let", "await", "static", "implements", "package",
+                "protected", "interface", "private", "public"
+            )
+        }
+
+        init {
+            require(identifierRegex.matches(name)) {
+                "'$name' is not valid JavaScript identifier"
+            }
+
+            require(name !in reservedKeywords) {
+                "'$name' is reserved keyword in JavaScript"
+            }
+        }
+
+        public val litItem: String get() = "\${$name}"
+    }
+
+    public operator fun String.invoke(provider: (TSource) -> String): Property<TSource>
+
+    public fun property(provider: (TSource) -> String): ReadOnlyProperty<Any?, Property<TSource>>
+
+    public fun function(name: String, handler: (TSource) -> Unit): Function<TSource>
+
+    public fun function(handler: (TSource) -> Unit): ReadOnlyProperty<Any?, Function<TSource>>
+
+    public fun templateExpression(templateExpression: String)
+
+    public fun templateExpression(initBlock: KLitRendererTagsBuilderA<TSource>.() -> Unit)
+
+}
+
+/**
+ * --- Renderers https://vaadin.com/docs/latest/components/grid/renderers
```

Comment:
Same thing: I'm quite lost as to what this interface is supposed to do, and the current kdoc (which is a list of links) isn't really helping.